doc/CounterDefinitions.markdown

```diff
@@ -0,0 +1,166 @@
+
+Calculated using log(Outbound/Inbound, 3), with the following boundary
+conditions:
+* 0.0 when Outbound equals Inbound,
+* -.inf when Outbound is zero, and
+* .inf when Inbound is zero.
+
+### TrafficModelEmissionCount
+
+The number of packets emitted by the relay.
+
+The traffic model assumes each packet is at most 1500 bytes. Byte events are
+split into packets, assuming no delay between emitted packets in the same byte
+event.
+
+### TrafficModelTransitionCount
+
+The number of state transitions in the traffic model.
+
+### TrafficModelLogDelayTime and TrafficModelSquaredLogDelayTime
+
+The (squared) natural logarithm of the inter-packet delay time in microseconds.
+These are derived from socket read/write timestamps, in a similar way to
+InterStreamCreationTimes. Each socket read/write is then split on 1500 byte
+boundaries into packets. The entire delay is assigned to the first packet in
+the socket read/write.
+
+Calculated using int(log(DelayTime)), with the following boundary
+conditions:
+* 0 when DelayTime is less than 1 microsecond.
+
+For the Squared counter, the value is squared after truncation to an integer.
+
+See LifeTime for information about timestamp and event loop precision.
+
+### ZeroCount
+
+A counter that is not incremented by any relay.
+
+The aggregated value of this counter must be zero. A non-zero value indicates
+that one or more Share Keepers or Data Collectors failed to provide results.
+
+## Counter Variants and Sub-Categories
+
+This section describes how some counters are split or filtered.
+
+### Entry/Exit
+
+This counter is only collected when the relay is in this position in the
+Connection or Circuit.
+
+For example:
+* When Tor clients connect to the Tor network, they make an Entry Connection.
+* When Tor clients access a remote server, they make an Exit Stream.
+
+### Active/Inactive
+
+An Active Circuit is being used by a Tor client. Circuit Activity is checked
+when a Circuit ends.
+
+The activity thresholds are:
+* Entry Circuits: 8 or more Cells were transmitted over the Circuit.
+                  Uses the sum of Inbound and Outbound Cells.
+* Entry ClientIPs: 1 or more Entry Circuits were active in this rotation
+                   period.
+* Exit Circuits: 1 or more Streams ended on the Circuit.
+
+### Stream Port
+
+Stream sub-categories based on the remote port used by the Stream. A Circuit
+can belong to multiple sub-categories based on its Streams: if it does,
+its counters are incremented for each sub-category.
+
+The sub-categories are:
+* Interactive: ports typically used for low-latency connections
+* P2P: ports typically used for bulk data transfer
+* Web: ports typically used for web browsing
+* OtherPort: any ports not listed in any other category
+
+### Inbound/Outbound
+
+The direction of Cell or Byte traffic on the Circuit or Stream.
+
+Inbound data is read/received/downloaded by the relay.
+Outbound data is written/sent/uploaded by the relay.
```

Comment:
It depends whether we are talking bytes or cells:

PrivCount increments OutboundByteCount using writebw: 
https://github.com/privcount/privcount/blob/master/privcount/data_collector.py#L735

These counts come from privcount_n_written:
https://github.com/privcount/tor/blob/privcount/src/or/control.c#L6643

Which are incremented in connection_handle_write_impl whenever tor writes bytes to a stream:
https://github.com/privcount/tor/blob/ee91d11bb45776bcd30efc0e1b3039cece5e5290/src/or/connection.c#L3977

PrivCount increments OutboundCellCount using ncellsout: 
https://github.com/privcount/privcount/blob/master/privcount/data_collector.py#L842

These counts come from privcount_n_cells_out:
https://github.com/privcount/tor/blob/privcount/src/or/control.c#L6712

Which is incremented in channel_flush_from_first_active_circuit whenever tor writes cells to the next (Outbound) hop:
https://github.com/privcount/tor/blob/ee91d11bb45776bcd30efc0e1b3039cece5e5290/src/or/relay.c#L2668

I'll clarify the documentation.